streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -901,6 +901,36 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
   )(f: (B, C) => D)(implicit ev: A00 =:= A1): ZSink[R1, E1, A00, A1, D] =
     zip(that).map(f.tupled)
 
+  final def andThen[R1 <: R, E1 >: E, A00 >: A0, C, A000](that: ZSink[R1, E1, A00, B, C]): ZSink[R1, E1, A0, A, C] =
```

Comment:
We should also add a non-IO version of this, overriden in `SinkPure`, for fusing together sinks that do not require IO effects.